src/java.base/share/classes/sun/util/locale/provider/LocaleServiceProviderPool.java

```diff
@@ -369,7 +369,10 @@ static Locale getLookupLocale(Locale locale) {
                         "A locale(" + locale + ") has non-empty extensions, but has illformed fields.");
 
                 // Fallback - script field will be lost.
-                lookupLocale = new Locale(locale.getLanguage(), locale.getCountry(), locale.getVariant());
+                lookupLocale = Locale.forLanguageTag(
+                        (locale.getLanguage().isEmpty() ? "und" : locale.getLanguage()) +
+                        (locale.getCountry().isEmpty() ? "" : "-" + locale.getCountry()) +
+                        (locale.getVariant().isEmpty() ? "" : "-x-lvariant-" + locale.getVariant()));
```

Comment:
It seems like this snippet (and ones very similar to it) are repeated several times throughout the JDK code as replacements for the two- and three-arg constructors. This seems like a fair increase in complexity, and the use of "und" and "-x-lvariant-" are quite non-obvious. Would we recommend that third party code that uses the Locale constructors replace them with this snippet? Is there something better that we can provide?

Reply:
True. One solution could be to expose `Locale.getInstance()`, which is currently a package-private static method, for this purpose. Though that means promoting `ill-formed` locale objects which defy some part of the deprecation cause.

Reply:
Introduced a new `Locale.of(String...)` method.